src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+            },
+        };
+
+        if (enable === undefined && samplingRate === undefined) {
+            return;
+        }
+
+        if (!twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME] = {};
+        }
+
+        if (enable !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_mode = enable ? 2 : 1;
+        }
+
+        if (samplingRate !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_rate = samplingRate;
+        }
+
+        if (deviceIds.length === 1) {
+            return new Promise((resolve, reject) => {
+                let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+                registry.updateTwin(deviceIds[0], JSON.stringify(twinPatch), twinPatch.etag, (err) => {
+                    if (err) {
+                        return reject(err.message);
+                    } else {
+                        return resolve("");
+                    }
+                });
+            });
+        }
+
+        return await this.updateDeviceTwinJob(twinPatch, iotHubConnectionString, deviceIds);
+    }
+
+    private updateDeviceTwinJob(twinPatch, iotHubConnectionString: string, deviceIds: string[]) {
+        return new Promise((resolve, reject) => {
+            let twinJobId = uuid.v4();
+            let jobClient = iothub.JobClient.fromConnectionString(iotHubConnectionString);
+
+            let queryCondition = this.generateQureyCondition(deviceIds);
+            let startTime = new Date();
+            let maxExecutionTimeInSeconds = 300;
+
+            jobClient.scheduleTwinUpdate(twinJobId,
+                queryCondition,
+                twinPatch,
+                startTime,
+                maxExecutionTimeInSeconds,
+                (err) => {
+                    if (err) {
+                        reject("Could not schedule distributed tracing setting update job: " + err.message);
+                    } else {
+                        this.monitorJob(twinJobId, jobClient, (e, result) => {
+                            if (e) {
+                                reject("Could not monitor distributed tracing setting update job: " + e.message);
+                            } else {
+                                resolve("Detailed information are shown as below:\n" + JSON.stringify(result, null, 2));
+                            }
+                        });
+                    }
+                });
+        });
+    }
+
+    private generateQureyCondition(deviceids: string[]): string {
+        const deviceIdsWithQuotes = deviceids.map((id) => "'" + id + "'");
+        return `deviceId IN [${deviceIdsWithQuotes.join(",")}]`;
+    }
+
+    private monitorJob(jobId, jobClient, callback) {
+        let jobMonitorInterval = setInterval(() => {
+            jobClient.getJob(jobId, (err, result) => {
```

Comment:
use promise. For new code, let's use promise if we can

Reply:
fixed